openstack_hypervisor/hooks.py

```diff
@@ -930,6 +932,376 @@ def _get_datapath_type(context: dict) -> str:
         return "system"
 
 
+def _get_dpdk_port_name(ifname: str) -> str:
+    """Get the DPDK port name for the specified interface."""
+    # Charmed Openstack used a hash of the pci address:
+    # https://github.com/juju/charm-helpers/blob/33c08fc064069c30237b47f1998ac9996351301a/charmhelpers/contrib/openstack/context.py#L2950
+    #
+    # Sice Sunbeam uses interface names, we'll go with dpdk-{ifname}.
+    return f"dpdk-{ifname}"
+
+
+def _get_dpdk_mappings(snap: Snap, context: dict) -> dict:
+    # We're using a snap setting to store information about
+    # DPDK ports, bridges and bonds.
+    #
+    # Note that once an interface is bound to the vfio-pci driver,
+    # it will no longer be visible to the host. At the same time,
+    # we're modifying the initial netplan configuration as part of
+    # the dpdk enablement.
+    mappings = (context.get("internal") or {}).get("dpdk_port_mappings") or {}
+
+    if isinstance(mappings, str):
+        mappings = json.loads(mappings)
+
+    if "ports" not in mappings:
+        mappings["ports"] = {}
+    if "bonds" not in mappings:
+        mappings["bonds"] = {}
+
+    logging.debug("Retrieved DPDK port mappings: %s", mappings)
+    return mappings
+
+
+def _set_dpdk_mappings(snap: Snap, mappings: dict):
+    logging.debug("Updated DPDK port mappings: %s", mappings)
+    snap.config.set({"internal.dpdk-port-mappings": json.dumps(mappings or {})})
+
+
+def _process_dpdk_ports(snap: Snap, context: dict):
+    ovs_dpdk_enabled = context.get("network", {}).get("ovs_dpdk_enabled")
+    dpdk_ifaces = (context.get("network", {}).get("ovs_dpdk_ports") or "").split(",")
+    dpdk_ifaces = [iface.strip() for iface in dpdk_ifaces if iface]
+    dpdk_driver = context.get("network", {}).get("dpdk_driver") or "vfio-pci"
+
+    if not ovs_dpdk_enabled:
+        logging.info("DPDK disabled, skipping physical port configuration.")
+        return
+    if not dpdk_ifaces:
+        logging.info("No DPDK interface specified, skipping physical port configuration.")
+        # For now, we'll just ignore interfaces that were removed
+        # from the list of dpdk ports.
+        return
+
+    logging.info("DPDK interface names: %s", dpdk_ifaces)
+
+    # Previously processed dpdk port mappings.
+    dpdk_mappings = _get_dpdk_mappings(snap, context)
+
+    _process_dpdk_netplan_config(dpdk_mappings, dpdk_ifaces)
```

Comment:
It may not be clear where the dpdk mappings get populated, should add an inline comment here.